tests/bson-deserialize-typewrapped-binary-001.phpt

```diff
@@ -0,0 +1,105 @@
+--TEST--
+BSON serialization: type wrapped Binary
+--FILE--
+<?php
+$tests = [
+	[ 'types' => [] ],
+	[ 'types' => [ 'Binary' => 'MyBinary' ] ],
+	[ 'root' => 'array', 'types' => [ 'Binary' => 'MyBinary' ] ],
+	[ 'types' => [ 'Binary' => 'MyBinaryReturnScalarType' ] ],
+];
+
+class MyBinary implements \MongoDB\BSON\TypeWrapper
+{
+	public $intern;
+
+	public function __construct(\MongoDB\BSON\Type $type)
+	{
+		$this->intern = $type;
+	}
+
+	static function createFromBSONType(\MongoDB\BSON\Type $type) : \MongoDB\BSON\TypeWrapper
+	{
+		if (! $type instanceof \MongoDB\BSON\Binary) {
+			echo "Not of the right type\n";
```

Comment:
Why not throw here? That's more consistent with what we'd expect users to do, and it would set a good example for them to follow.